Util/Xamarin.Build.TypeRedirector/source/Xamarin.Build.TypeRedirector/RedirectMovedTypes.cs

```diff
@@ -14,20 +14,34 @@ public class RedirectMovedTypes : Task
         [Required]
         public ITaskItem[] Assemblies { get; set; }
 
+        public ITaskItem[] References { get; set; }
+
         public override bool Execute()
         {
             foreach (var assemblySpec in Assemblies)
             {
                 var assembly = assemblySpec.ItemSpec;
-
-                if (!File.Exists(assembly))
+                if (string.IsNullOrEmpty(assembly) || !File.Exists(assembly))
                     continue;
 
+                Log.LogMessage($"Redirecting types for {assembly}...");
+
                 var tempFile = assembly + ".temp.dll";
 
+                using var resolver = new AssemblyResolver();
```

Comment:
No idea why my sample projects worked without this... Real projects require even the references to the System and Xamarin.iOS assemblies. Magic machine.